### DISCUSSION TITLE
Invariant Violation: View config getter callback for component `RNBaseCore` must be a function (received `undefined`).

### DISCUSSION THREAD

**Discussion by ayand269** [reactwg/react-native-new-architecture#173] [Q&A]
Hello Everyone,
I am an intermediate in React Native. I am trying to create a library with Fabric for learning purpose. But I stuck.
[This is my library](https://github.com/ayand269/react-native-base/tree/master/core)

And I use it like this
![Screenshot from 2024-04-17 01-32-48](https://github.com/reactwg/react-native-new-architecture/assets/59437316/715c2770-9d01-416b-a60f-20a7c8a51a7b)

And I got this error
![Screenshot from 2024-04-17 01-30-31](https://github.com/reactwg/react-native-new-architecture/assets/59437316/572f0325-2dd5-4ad1-9f3d-06f6fb589d24)

Can any one help me?

**Comment by satya164** [comment]
Shouldn't your view manager be named `RNBaseCoreViewManager`?

**Reply by ayand269** [reply]
Is it necessary? I register the view manager class in my `RNBaseCorePackage.java`

```java
@Override
public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
    return Collections.singletonList(new RNBaseCoreTextInputManager(reactContext));
}
```
So it should take that class as my view manager.

And also after changing the name it's give the same error.
